ProjFS.Mac/PrjFSKext/KauthHandler.cpp

```diff
@@ -21,6 +21,12 @@
 #include "KauthHandlerTestable.hpp"
 #endif
 
+enum ProviderCalloutUserPolicy
```

Comment:
Naming nit: Would `ProviderHydrationPolicy` be a clearer name? I think the use of callout here is uninituitive (to me at least).

Reply:
The reason I didn't go for that kind of name is because for example when sending `MessageType_KtoU_NotifyDirectoryPreDelete` and `MessageType_KtoU_NotifyFilePreDelete` we need to use `UserPolicy_AllowAny` and this has nothing to do with hydration.

"ProviderRequest" doesn't make sense either, as again notifications aren't really requests, and "ProviderNotification" isn't quite right for hydration/enumeration requests. So I went with Callout because it's about the most generic word I could think of for making a roundtrip to the provider, be it for hydration/enumeration or notifications. I don't love the name either, so I'll easily be persuaded to change it. :-)